qiskit_experiments/test/IQPulseBackend.py

```diff
@@ -0,0 +1,84 @@
+            run_inputs = [run_inputs]
+            schedules = []
+            for circuit in run_inputs:
+                # divide the case in two 
+                # when circuit has it's own calibration or no calibration
+                if len(circuit.calibrations) == 0: # No cal -> extract the schedule directly
+                    schedule = schedule(circuit, self)# self position is for the backend : is self right? 
+                    # How can I give the backend as an argument??
+                    schedules.append(schedule)
+                elif circuit.calibrations : # eventhough there's cal, directly extracted schedule include that cal
+                    schedule = schedule(circuit, self)
+                    schedules.append(schedule)
+                # simulate the circuit's extracted schedult with the solver
+                solver = Solver(
+                    static_hamiltonian = self.hamiltonian_static,
+                    hamiltonian_operators = self.hamiltonian_operator,
+                    rotating_frame = self.hamiltonian_static,
+                    rwa_cutoff_freq=2*5.0,
+                    hamiltonian_channels = ['d0'] # how to define what qubit to use?
+                    channel_carrier_freqs = {'d0' : w}, #how to define what qubit to use?
+                    dt = 0.222
+                    )
+                # gave a specific number just for convenience for the time
+                sol = solver.solve(t_span=[0., 20.], y0=Statevector([1., 0.]), 
+                                    signals = schedules, t_eval=np.linspace(0., 200, 2000))
+###################################################################################
+# This is the most confusing part. I dont understand the necessity of this part.
+# Isnt schedule(circuit, backend) enough for extracting the pulse shcedule??
+# Why calibration.items?? I checked If the experiment include calibration, the generated circuit reflect the calibration already.              
+        circuit_unitaries = {}
+        circuit.add_calibration()
+        for inst_name, instructions in circuit.calibrations.items():
+            circuit_unitraires[inst_name] = instructions
+            for qubits_params, schedule in instructions.items():       
+                
+                unitary = self._run_pulse_simulation(schedule)#Does it mean Solver.solve?? or the real pulse simulator??
+                 
+                circuit_unitaries[inst_name][qubits_params] = unitary
+         # multiply the unitaries for the circuit
+        psi = np.array([[1], [0], [0]])  # Need to be a bit more clever here with the dimensions
+        for inst in circuit.data:  # TODO check
+            qubits, params, inst_name = self._get_info(inst)  # TODO
+                
+            unitary = circuit_unitaries[inst_name][(qubits, params)]
+                
+            psi = unitary @ psi  # Forward unitary dynamics.
+
+
+
+
+            return FakeJob(self, Result.from_dict(result))
```

Comment:
This part is still missing the conversion from state vector to e.g. IQ data or counts depending on the desired output. Can you add the following after the loop over the circuit data:
```python
result = self._statevector_to_result(psi, meas_return, meas_level)
```
Then I also add the following method outside of the `run` method:
```python
from qiskit.quantum_info import Statevector
from qiskit.qobj.utils import MeasLevel, MeasReturnType

def _state_vector_to_result(self, state: Statevector, meas_return: MeasReturnType, meas_level: MeasLevel):
    """Convert the state vector to IQ data or counts."""
    if meas_level == MeasLevel.CLASSIFIED:
        counts = ... # sample from the state vector. There might already be functions in Qiskit to do this.
    if meas_level == MeasLevel.KERNELED:
        iq_data = ... create IQ data.
```
We will then need to collect the data in a dict like `MockIQBackend` currently does so that we can convert it to a valid Qiskit Result object with the `from_dict` method.